api/src/main/java/org/eclipse/microprofile/openapi/annotations/callbacks/Webhook.java

```diff
@@ -0,0 +1,68 @@
+/*
+ * Copyright (c) 2024 Contributors to the Eclipse Foundation
+ * <p>
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except in compliance with the License. You may obtain a
+ * copy of the License at
+ * <p>
+ * http://www.apache.org/licenses/LICENSE-2.0
+ * <p>
+ * Unless required by applicable law or agreed to in writing, software distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the License for the specific language governing permissions and limitations
+ * under the License.
+ */
+package org.eclipse.microprofile.openapi.annotations.callbacks;
+
+import java.lang.annotation.Retention;
+import java.lang.annotation.RetentionPolicy;
+import java.lang.annotation.Target;
+
+import org.eclipse.microprofile.openapi.annotations.extensions.Extension;
+import org.eclipse.microprofile.openapi.models.PathItem;
+
+/**
+ * Describes a named webhook which the API may be instructed to call.
+ * <p>
+ * Webhooks are similar to {@link Callback callbacks} but are set up via an out-of-band mechanism (e.g. an admin web
+ * page) rather than by a call to an operation.
+ *
+ * @since 4.0
+ */
+@Target({})
+@Retention(RetentionPolicy.RUNTIME)
+public @interface Webhook {
```

Comment:
Should this also include `parameters` (applicable to all operations) and `servers` [1] since this is effectively a path item? I wonder also if it should have a `$ref` for the same reason.

[1] https://spec.openapis.org/oas/v3.1.0.html#fixed-fields-6

Reply:
> Should this also include `parameters` and `servers`

It _could_, but I'm not sure whether we should or not. Path items appear in four places: under `paths`, `callbacks`, `webhooks` and `components` (not yet added).

By far the most common is under `paths` and we don't currently have any way to add `parameters` or `servers` there with annotations since there's nothing in Jakarta REST that really maps to path item and so there's nowhere to put an annotation.

`@Callback` also represents a path item, so if we were to add `parameters` and `servers` to `@Webhook`, we should probably add them to `@Callback` as well.

> I wonder also if it should have a `$ref` for the same reason.

Yes possibly. I was going to think about that under #437.

Another thought occurred to me: if we were to add a new annotation for #437, it would probably have identical fields to `@Webhook` (since both represent one entry in a `Map<String, PathItem>`). We could use one annotation (named `@PathItem`) for both instead.

Reply:
Let's design this alongside #437.

I'll remove the annotations from this PR so we can get the model parts merged.

Reply:
Annotations removed.